### PR TITLE
Less strict docker dependency versions

### PR DESCRIPTION
**Pull request by jcugat**
# TL;DR
Change the minimum required version of the docker dependency from v5.* to v4.*

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
We need it because we use Flyte alongside other packages which force a docker version in the v4.* range (specifically tensorflow's [tfx](https://github.com/tensorflow/tfx/blob/c29b35a3d1ee33076619958fc88831f5ef2c48ef/tfx/dependencies.py#L64) package).

I have tried running the tests locally while force-installing docker `docker==4.0.2` and it worked without any errors, so there's no breaking change between those versions that would affect this project.

Also it would be great if this change was backported to the Flyte v1.2.* and v1.3.* branches, thanks!

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pingsutw** [comment]
@jcugat I just curious how do you integration tfx with flyte. are you compiling tfx workflow to flyte workflow?

**Comment by jcugat** [comment]
@pingsutw Not exactly, we use Flyte to launch Kubeflow Pipelines, which uses TFX under the hood. There's some more details in this blog post: https://engineering.atspotify.com/2019/12/the-winding-road-to-better-machine-learning-infrastructure-through-tensorflow-extended-and-kubeflow/

**Review by pingsutw — APPROVED** [review]

**Comment by jcugat** [comment]
Thanks for merging @pingsutw!

Would it be possible to also backport this to the v1.2 and v1.3 releases? Should I do anything to help that happen?
